optimizer_wrapper.rb

```diff
@@ -927,15 +927,15 @@ def self.zip_cluster(vrp, cluster_threshold, force_cluster)
             # find intersection with tw of service_tw
             compatible_tws = service_tw.select{ |tw|
               tw[:day_index].nil? || new_tw[:day_index].nil? || tw[:day_index] == new_tw[:day_index] &&
-                (tw[:start].nil? || new_tw[:end].nil? || tw[:start] <= new_tw[:end]) &&
-                (tw[:end].nil? || new_tw[:start].nil? || tw[:end] >= new_tw[:start])
+                (new_tw[:end].nil? || tw[:start] <= new_tw[:end]) &&
```

Comment:
```suggestion
                (new_tw.end.nil? || tw.start <= new_tw.end) &&
                (tw.end.nil? || tw.end >= new_tw.start)
```
Could you also edit the rest of the function ?